core/idl_types.h

```diff
@@ -19,7 +19,11 @@
 
 #include "core/idl_base.h"
 
+struct IDLBoolean final : public IDLBaseHelper<bool> {};
 struct IDLDouble final : public IDLBaseHelper<double> {};
+struct IDLFloat final : public IDLBaseHelper<float> {};
+struct IDLLong final : public IDLBaseHelper<long> {};
```

Comment:
According to the [spec](https://heycam.github.io/webidl/#idl-long), the IDLLong type is a signed integer type that has values in the range [−2147483648, 2147483647].
BTW, the long type (in C++) might not be the same range. In other words, it can has different ranges in each different platforms.
Therefore, you should use explicit types such as int32_t, int64_t.

Reply:
I checked platform when return value.
Could you review it?